src/Microsoft.AspNetCore.Authorization/IAuthorizationEvaluator.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Authorization
+{
+    /// <summary>
+    /// Determines whether an authorization request was successful or not.
+    /// </summary>
+    public interface IAuthorizationEvaluator
+    {
+        /// <summary>
+        /// Returns true, if authorization has failed.
+        /// </summary>
+        /// <param name="context">The authorization information.</param>
+        /// <returns>True if authorization has failed.</returns>
+        bool HasFailed(AuthorizationHandlerContext context);
```

Comment:
Not sure that I see the value of having HasSucceeded and HasFailure especially given that I only see usage of HasSucceeded. HasFailure should always be ! HasSucceeded. So simplify the contract and remove the method. Or if the method must stay then make it an extension method to force the implementation.